Backpack/PageIndicator/Classes/BPKPageIndicator.swift

```diff
@@ -0,0 +1,91 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2023 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+public final class BPKPageIndicator: UIView {
+    public enum Variant {
+        case `default`, overImage
+    }
+    
+    private let pageControl = UIPageControl()
+    
+    //    This sets the current highlighted indicator
+    public var currentIndex: Int {
+        didSet {
+            pageControl.currentPage = currentIndex
+        }
+    }
+    
+    //    This is the total number of indicators to be available
+    public var totalIndicators: Int {
+        didSet {
+            pageControl.numberOfPages = totalIndicators
+        }
+    }
+    
+    //    Switches between default and overImage style
+    public let variant: Variant
+    
+    public init(
+        variant: Variant = .default,
+        currentIndex: Int,
+        totalIndicators: Int
+    ) {
+        self.variant = variant
+        self.currentIndex = currentIndex
+        self.totalIndicators = totalIndicators
+        
+        super.init(frame: .zero)
+        setupPageControl()
+        setupConstraints()
+    }
+    
+    required init?(coder: NSCoder) {
```

Comment:
Please add @available(*, unavailable) to this init.

Reply:
Added it in new commit